src/main/java/org/spongepowered/api/map/color/MapColor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.color;
+
+import com.google.common.base.Optional;
+import org.spongepowered.api.util.Color;
+
+public interface MapColor {
+
+    /**
+     * Returns the corresponding {@link Color} for this {@link MapColor}.
+     *
+     * @return The RGB color of this map color
+     */
+    Color getColor();
+
+    /**
+     * Returns the value of the entry in a map palette, this isn't the base
+     * color index which needs to be modified before being set, this represents
+     * the actual value that can be set in a map.
+     *
+     * @return The map color index
+     */
+    int getPaletteIndex();
```

Comment:
Efficiency, some plugins might wish to do very rapid map updates, possible API addition for setting pixels directly. Will add to Todo list possible API additions, this would be removed if no such feature is deemed required.